### PR TITLE
Fix basic auth not working for migrated APIs after API update

### PR DESCRIPTION
**Pull request by Tharanidk**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package org.wso2.carbon.apimgt.api.model;
+
+        import org.junit.Assert;
```
Comment (dushaniw) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java:3]: formatting issue

Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java:3]: license header is missing

Comment (Tharanidk) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java]: I have added the changes

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/API.java [hunk 2]
```diff
@@ -904,6 +912,69 @@ public String getEndpointConfig() {
...
+                    } else {
+                        productionAndSandbox.setType(APIConstants.ENDPOINT_SECURITY_TYPE_BASIC);
+                    }
+                    productionAndSandbox.setUsername(getEndpointUTUsername());
+                    productionAndSandbox.setPassword(getEndpointUTPassword());
+                    Object productionAndSandboxSecurity = parser.parse(
+                            objectMapper.writeValueAsString(productionAndSandbox));
+                    epSecurity.put(APIConstants.ENDPOINT_SECURITY_PRODUCTION, productionAndSandboxSecurity);
+                    epSecurity.put(APIConstants.ENDPOINT_SECURITY_SANDBOX, productionAndSandboxSecurity);
+                    endpointConfigJson.put(APIConstants.ENDPOINT_SECURITY, epSecurity);
```
Comment (madushadhanushka) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/API.java:936]: This seems to be repetitive code throughout if else code blocks. Can we get this out of the if condition?

File: components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (madushadhanushka) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java:2]: Old licence

File: components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        JSONObject sandboxEndpoint = (JSONObject) ((JSONObject) endpointConfigJson.get("endpoint_security")).get("sandbox");
+
+        Assert.assertNotNull("Endpoint Security not available for Production Endpoints" ,
+                endpointConfigJson.containsKey("production_endpoints"));
+        Assert.assertNotNull("Endpoint Security not available for Sandbox Endpoints" ,
+                endpointConfigJson.containsKey("sandbox_endpoints"));
+
+        Assert.assertTrue("Endpoint Security not enabled for Production Endpoints",
+                (Boolean) productionEndpoint.get("enabled"));
+
```
Comment (shilmyhasan) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java:63]: formatiing issues

File: components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    public void init() throws Exception {
+        apiIdentifier = new APIIdentifier("P1_API1_v1.0.0");
+        api = new API(apiIdentifier);
+        api.setEndpointSecured(true);
+    }
+
+    @Test
+    public void testEndpointConfigurationsSecurity() throws Exception {
+        String endpointConfig = "{\"production_endpoints\":{\"url\":\"https://webhook.site/4e659833-6181-43b6-9df6-02787e5054a0\",\"config" +
+                "\":null,\"template_not_supported\":false},\"sandbox_endpoints\":{\"url\":\"https://webhook.site/4e659833-6181-43b6-9df6-02787e5054a0" +
```
Comment (shilmyhasan) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java:43]: formatting issues, line length should not exceed

File: components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java (hunk 3 above)
Comment (Tharanidk) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java:2]: corrected

File: components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java (hunk 4 above)
Comment (Tharanidk) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java:63]: Done

File: components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java (hunk 5 above)
Comment (Tharanidk) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/test/java/org/wso2/carbon/apimgt/api/model/APITest.java:43]: Done

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/API.java (hunk 2 above)
Comment (Tharanidk) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/API.java:936]: These if-else represent three separate scenarios and the object the value set is being varied as per the scenario.

**Review by tharikaGitHub — APPROVED** [review]

**Pull request merged**